blazor-application/Services/BLEScannerService.cs

```diff
@@ -0,0 +1,93 @@
+using BlazorBLE.Data;
+using Plugin.BLE;
+using Plugin.BLE.Abstractions.Contracts;
+using Plugin.BLE.Abstractions.EventArgs;
+using Plugin.BLE.Abstractions.Exceptions;
+
+namespace BlazorBLE.Services;
+
+public sealed class BLEScannerService
+{
+    public event Action DevicesChanged;
+
+    public IReadOnlyList<IDevice> Devices => devices;
+
+    private readonly List<IDevice> devices = new();
+    private readonly IAdapter adapter;
+    private readonly IPromptService promptService;
+
+    public BLEScannerService(IPromptService promptService)
+    {
+        this.promptService = promptService;
+
+        adapter = CrossBluetoothLE.Current.Adapter;
+        adapter.ScanMode = ScanMode.LowLatency;
+        adapter.DeviceDiscovered += Adapter_DeviceDiscovered;
+    }
+
+    ~BLEScannerService()
+    {
+        adapter.DeviceDiscovered -= Adapter_DeviceDiscovered;
+        adapter.StopScanningForDevicesAsync();
+    }
+
+    private void Adapter_DeviceDiscovered(object sender, DeviceEventArgs args)
+    {
+        IDevice device = args.Device;
+
+        if (!KBeaconData.IsProximityBeacon(device)) return;
+
+        Console.WriteLine($"Discovered proximity beacon: {device.Id} {device.Name}");
+
+        devices.Add(device);
+        devices.Sort((deviceA, deviceB) => deviceB.Rssi - deviceA.Rssi);
+
+        DevicesChanged?.Invoke();
+    }
+
+    public async void BeginScan()
+    {
+        if (!await AppPermissions.RequestPermissions()) 
```

Comment:
Do we request every time? Would it be better to store whether permission is granted on the instance? I do not know the context of class usage, but it seems weird to request permission every time a scanning begins

Reply:
I have been corrected. This just reads wierd.